src/NServiceBus.Metrics/MetricsOptions.cs

```diff
@@ -65,14 +65,16 @@ public void EnableCustomReport(Func<string, Task> func, TimeSpan interval)
         /// </summary>
         /// <param name="serviceControlMetricsAddress">The transport address of the ServiceControl instance</param>
         /// <param name="interval">Interval between consecutive reports</param>
+        /// <param name="instanceId">Unique identifier of endpoint physical instance</param>
         [ObsoleteEx(Message = "Not for public use.")]
-        public void SendMetricDataToServiceControl(string serviceControlMetricsAddress, TimeSpan interval)
+        public void SendMetricDataToServiceControl(string serviceControlMetricsAddress, TimeSpan interval, string instanceId)
```

Comment:
What I mean is that in general host identifier is not enough and scenario that you mentioned (multiple physical instances running on the same machine is example of that). 

> It seems to me host identifier from core would be enough?
Under the assumption that users would remember to always set it up properly. With additional parameter we are making sure that always happens.